pkg/authorizationserver/issuermetadata.go

```diff
@@ -0,0 +1,28 @@
+package authorizationserver
+
+import (
+	"context"
+	"net/http"
+	"os"
+
+	"github.com/TBD54566975/ssi-sdk/oidc/issuance"
+	"github.com/goccy/go-json"
+	"github.com/tbd54566975/ssi-service/pkg/server/framework"
+)
+
+// Allow retrieval of credential issuer metadata according to https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html#name-credential-issuer-metadata
+func credentialIssuerMetadata(config *AuthConfig) func(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
+	// It's ok to panic inside this function because this is only called during startup.
+	jsonData, err := os.ReadFile(config.CredentialIssuerFile)
```

Comment:
if this is the same file being retrieved over and over we can cache it in memory?

Reply:
It is actually cached! This function is only run once, during the setup of the server. If you look closely, this function returns _another function_. The latter uses the already read file.